app/routes/products.$categorySlug/route.tsx

```diff
@@ -62,72 +81,148 @@ export const handle: RouteHandle<typeof loader> = {
 };
 
 export default function ProductsPage() {
-    const { category, categoryProducts, allCategories } = useLoaderData<typeof loader>();
+    const { category, categoryProducts, allCategories, productPriceBounds } =
+        useLoaderData<typeof loader>();
+
     const breadcrumbs = useBreadcrumbs();
 
+    const { filters, someFiltersApplied, applyFilters, clearFilters, clearAllFilters } =
+        useProductFilters();
+
+    const currency = categoryProducts.items[0]?.priceData?.currency;
+
+    const priceFormatter = useMemo(
```

Comment:
How about adding a general-purpose utility `formatPrice(price, currency)` that can be imported directly by any component that wants to format the price?